authentication/auth.go

```diff
@@ -558,6 +556,32 @@ func GenerateTokenFromDefaultsV2(schemaURL string, accountServiceURL string, url
 	return token, ""
 }
 
+func addAdditionalMetadata(surveyMetadata map[string]interface{}, requiredSchemaMetadata []Metadata) []Metadata {
```

Comment:
Do we need the nested loops here if we are just updating the map with any quick launch bool values?

Reply:
Updated in new PR